### PR TITLE
PHPCS: JSON GET endpoints

### LINKED ISSUE #21955: PHPCS: JSON Endpoints get
### Impacted plugin

Jetpack

### What

PHPCS

### How

```
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comment-counts-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comment-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comment-history-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comments-tree-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comments-tree-v1-1-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comments-tree-v1-2-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-customcss.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-media-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-media-v1-1-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-media-v1-2-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-counts-v1-1-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-v1-1-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-v1-2-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-taxonomies-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-taxonomy-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-term-endpoint.php",
```

### PR DESCRIPTION
**Pull request by sdixon194**
Fixes #21955

#### Changes proposed in this Pull Request:

* Fixes JSON Get endpoints.

Most are formatting changes, but some things I ignored that a gut check would be good one:

* class.wpcom-json-api-get-post-counts-v1-1-endpoint.php - ignored some database related phpcs complaints I didn't know about.
* projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php - ignored the only one object structure allowed in a file, since splitting things up might be a headache right now.

#### Jetpack product discussion

n/a
#### Does this pull request change what data or activity we track or use?

no
#### Testing instructions:

* Proof? Is it possible to test the endpoints?

### CODE REVIEW

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comments-tree-endpoint.php [hunk 1]
```diff
@@ -140,22 +147,26 @@ function get_comment_db_status( $status ) {
...
-	 * @param string $status
+	 * @param string $status - comment status.
 	 *
 	 * @return boolean
 	 */
-	function validate_status_param( $status ) {
+	public function validate_status_param( $status ) {
+		// @todo see if we can use a strict comparison here.
+		// phpcs:ignore WordPress.PHP.StrictInArray.MissingTrueStrict
 		return in_array( $status, array( 'all', 'approved', 'unapproved', 'pending', 'spam', 'trash' ) );
```
Comment (kraftbj) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comments-tree-endpoint.php:157]: We should be able to add a true here. We do want the `$status` to match one of those, so we do want to compare a string to a string.

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comments-tree-endpoint.php [hunk 2]
```diff
@@ -140,22 +147,26 @@ function get_comment_db_status( $status ) {
...
-	 * @param string Comment type: can be 'trackback', 'pingback', or 'comment'.
+	 * @param string $type Comment type: can be 'trackback', 'pingback', or 'comment'.
 	 *
 	 * @return string Sanitized comment type.
 	 */
-	function get_sanitized_comment_type( $type = 'comment' ) {
+	public function get_sanitized_comment_type( $type = 'comment' ) {
+		// @todo see if we can use a strict comparison here.
+		// phpcs:ignore WordPress.PHP.StrictInArray.MissingTrueStrict
 		if ( in_array( $type, array( 'trackback', 'pingback', 'comment' ) ) ) {
```
Comment (kraftbj) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-comments-tree-endpoint.php:170]: same, let's true it.

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-counts-v1-1-endpoint.php [hunk 3]
```diff
@@ -1,102 +1,134 @@
...
+	 * @param int    $id - the ID.
+	 */
+	private function retrieveCounts( $post_type, $id = null ) {
 		if ( ! isset( $id ) ) {
 			$counts = array();
-			foreach( (array) wp_count_posts( $post_type ) as $status => $count ) {
+			foreach ( (array) wp_count_posts( $post_type ) as $status => $count ) {
+				// @todo see if we can use a strict comparison here.
+				// phpcs:ignore WordPress.PHP.StrictInArray.MissingTrueStrict
 				if ( in_array( $status, $this->whitelist ) && $count > 0 ) {
```
Comment (kraftbj) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-counts-v1-1-endpoint.php:82]: can true this too.

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-counts-v1-1-endpoint.php [hunk 4]
```diff
@@ -1,102 +1,134 @@
...
+	 *
+	 * @param array $in - the post we're checking.
+	 */
 	private function filterStatusesByWhiteslist( $in ) {
 		$return = array();
-		foreach( $in as $result) {
+		foreach ( $in as $result ) {
+			// @todo see if we can use a strict comparison here.
+			// phpcs:ignore WordPress.PHP.StrictInArray.MissingTrueStrict
 			if ( in_array( $result->status, $this->whitelist ) ) {
```
Comment (kraftbj) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-counts-v1-1-endpoint.php:115]: true too.

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-counts-v1-1-endpoint.php [hunk 5]
```diff
@@ -108,6 +140,8 @@ public function callback( $path = '', $blog_id = 0, $post_type = 'post' ) {
 			return $blog_id;
 		}
 
+		// @todo see if we can use a strict comparison here.
+		// phpcs:ignore WordPress.PHP.StrictInArray.MissingTrueStrict
 		if ( ! in_array( $post_type, array( 'post', 'revision', 'page', 'any' ) ) && defined( 'IS_WPCOM' ) && IS_WPCOM ) {
```
Comment (kraftbj) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-counts-v1-1-endpoint.php:145]: yeah let's true this one too.

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-v1-1-endpoint.php [hunk 6]
```diff
@@ -57,13 +73,15 @@ function callback( $path = '', $blog_id = 0, $post_id = 0 ) {
 		}
 
 		if ( defined( 'IS_WPCOM' ) && IS_WPCOM &&
+				// @todo see if we can use a strict comparison.
+				// phpcs:ignore WordPress.PHP.StrictInArray.MissingTrueStrict
 				! in_array( get_post_type( $post_id ), array( false, 'post', 'revision' ) ) ) {
```
Comment (kraftbj) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-post-v1-1-endpoint.php:78]: and this one

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 7]
```diff
@@ -788,42 +890,54 @@ function callback( $path = '', $blog_id = 0 ) {
 		$all_formats = get_post_format_strings();
 		$supported   = get_theme_support( 'post-formats' );
 
-		$supported_formats = $response['formats'] = array();
+		$supported_formats = $response['formats'] = array(); //phpcs:ignore
```
Comment (kraftbj) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php:893]: I believe we'd need to `$response = array()` before this to set it up, which should account for this line and L901 below. I don't recall if this is strictly a PHPCS issue or if PHP will start throwing a notice in a future version for this type of variable assumption.

**Comment by kraftbj** [comment]
A few suggestions. I need to dig in regarding the db issues; I don't remember off the top of my head the nuance around prepare, so no comment on those for the moment.

**Comment by sdixon194** [comment]
Those other issues should be addressed now!

**Review by kraftbj — APPROVED** [review]

**Comment by sdixon194** [comment]
Deployed to wpcom r240303-wpcom

**Comment by sdixon194** [comment]
r242398-wpcom finally

**Pull request merged**